bundles/org.openhab.binding.mpd/src/main/java/org/openhab/binding/mpd/internal/protocol/MPDConnectionThread.java

```diff
@@ -147,38 +146,45 @@ private void sendNoIdleIfInIdle() {
             try {
                 sendCommand(new MPDCommand("noidle"));
             } catch (IOException e) {
-                logger.debug("sendCommand(noidle) failed");
+                logger.debug("sendCommand(noidle) failed", e);
             }
         }
     }
 
-    private void establishConnection() throws IOException, MPDException {
+    private void establishConnection() throws UnknownHostException, IOException, MPDException {
         openSocket();
 
         MPDCommand currentCommand = new MPDCommand("connect");
         MPDResponse response = readResponse(currentCommand);
 
         if (!response.isOk()) {
-            throw new MPDException("could not connect");
+            throw new MPDException("Failed to connect to " + this.address + ":" + this.port);
         }
 
         if (!password.isEmpty()) {
             currentCommand = new MPDCommand("password", password);
             sendCommand(currentCommand);
             response = readResponse(currentCommand);
             if (!response.isOk()) {
-                throw new MPDException("could not authenticate");
+                throw new MPDException("Could not authenticate, please validate your password");
             }
         }
     }
 
-    private void openSocket() throws IOException {
+    private void openSocket() throws UnknownHostException, IOException, MPDException {
         logger.debug("opening connection to {} port {}", address, port);
+
+        if (address.isEmpty()) {
+            throw new MPDException("Missing parameter ipAddress");
+        }
+        if (port < 1 || port > 65335) {
+            throw new MPDException("Invalid parameter port");
```

Comment:
Very minor: 
```suggestion
            throw new MPDException("Invalid port parameter");
```

Otherwise, it might suggest that the parameter port is invalid and should not be supplied.

Reply:
@stefanroellin please let me know if you want to fix this tiny comment otherwise I'm fine merging as is, thanks for your work!

Reply:
I will push a fix for it.